contracts/DEXes/UniV3StablesJoint.sol

```diff
@@ -0,0 +1,639 @@
+
+        // Initialize to the current status of owed tokens
+        (_amountPending[0], _amountPending[1]) = tokenA < tokenB
+            ? (positionInfo.tokensOwed0, positionInfo.tokensOwed1)
+            : (positionInfo.tokensOwed1, positionInfo.tokensOwed0);
+
+        // Gas savings
+        IUniswapV3Pool _pool = IUniswapV3Pool(pool);
+        int24 _minTick = minTick;
+        int24 _maxTick = maxTick;
+
+        // Use Uniswap views library to calculate the fees earned in tokenA and tokenB based
+        // on current status of the pool and provided position
+        (uint128 tokensOwed0, uint128 tokensOwed1) = UniswapHelperViews.getFeesEarned(
+            UniswapHelperViews.feesEarnedParams(
+                positionInfo.liquidity,
+                _pool.slot0().tick,
+                _minTick,
+                _maxTick,
+                _pool.feeGrowthGlobal0X128(),
+                _pool.feeGrowthGlobal1X128(),
+                positionInfo.feeGrowthInside0LastX128,
+                positionInfo.feeGrowthInside1LastX128
+            ),
+            _pool.ticks(_minTick),
+            _pool.ticks(_maxTick)
+        );
+
+        // Reorder to make sure amounts are added correctly
+        if (tokenA < tokenB) {
+            _amountPending[0] += tokensOwed0;
+            _amountPending[1] += tokensOwed1;
+        } else {
+            _amountPending[1] += tokensOwed0;
+            _amountPending[0] += tokensOwed1;
+        }
+
+        return _amountPending;
+    }
+
+    /*
+     * @notice
+     *  Function called by the uniswap pool when minting the LP position (providing liquidity),
+     * instead of approving and sending the tokens, uniV3 calls the callback imoplementation
+     * on the caller contract
+     * @param amount0Owed, amount of token0 to send
+     * @param amount1Owed, amount of token1 to send
+     * @param data, additional calldata
+     */
+    function uniswapV3MintCallback(
+        uint256 amount0Owed,
+        uint256 amount1Owed,
+        bytes calldata data
+    ) external {
+        // Only the pool can use this function
+        require(msg.sender == pool); // dev: callback only called by pool
+        // Send the required funds to the pool
+        IUniswapV3Pool _pool = IUniswapV3Pool(pool);
+        IERC20(_pool.token0()).safeTransfer(address(_pool), amount0Owed);
+        IERC20(_pool.token1()).safeTransfer(address(_pool), amount1Owed);
+    }
+
+    /*
+     * @notice
+     *  Function called by the uniswap pool when swapping,
+     * instead of approving and sending the tokens, uniV3 calls the callback imoplementation
+     * on the caller contract
+     * @param amount0Delta, amount of token0 to send (if any)
+     * @param amount1Delta, amount of token1 to send (if any)
+     * @param data, additional calldata
+     */
+    function uniswapV3SwapCallback(
+        int256 amount0Delta,
+        int256 amount1Delta,
+        bytes calldata data
```

Comment:
same comment about requiring data to be empty just in case

Reply:
but it is not used anywhere.. so I can't come up with a way to exploit it. maybe not even worth it

Reply:
Discussing in MintCallback to resolving this